tests/unit_tests/layout/test_array_layout.py

```diff
@@ -424,6 +461,44 @@ def run(corsika_dict):
             file_name="file_doesnt_exist.yml"
         )
 
+    _save_db_config = array_layout_north_instance.mongo_db_config
+    array_layout_north_instance.mongo_db_config = None
+    with pytest.raises(ValueError):
+        corsika_dict = array_layout_north_instance._from_corsika_file_to_dict(
+            file_name=corsika_config_file
+        )
+    array_layout_north_instance.mongo_db_config = _save_db_config
+    array_layout_north_instance.site = None
+    with pytest.raises(ValueError):
+        corsika_dict = array_layout_north_instance._from_corsika_file_to_dict(
+            file_name=corsika_config_file
+        )
+
+
+def test_initialize_sphere_parameters():
+    w_1 = {"corsika_sphere_radius": 16.0 * u.m}
+    t_1 = ArrayLayout._initialize_sphere_parameters(w_1)
+    assert pytest.approx(t_1["corsika_sphere_radius"].value, 0.1) == 16.0
+    assert t_1["corsika_sphere_radius"].unit == u.m
+
+    w_2 = {"corsika_sphere_radius": "16. m"}
+    t_2 = ArrayLayout._initialize_sphere_parameters(w_2)
+    assert pytest.approx(t_2["corsika_sphere_radius"].value, 0.1) == 16.0
+    assert t_2["corsika_sphere_radius"].unit == u.m
+
+    w_3 = {"corsika_sphere_radius": {"value": 16.0, "unit": "m"}}
+    t_3 = ArrayLayout._initialize_sphere_parameters(w_3)
+    assert pytest.approx(t_3["corsika_sphere_radius"].value, 0.1) == 16.0
+    assert t_3["corsika_sphere_radius"].unit == u.m
+
+    w_4 = {"corsika_sphere_radius": 16.0}
+    t_4 = ArrayLayout._initialize_sphere_parameters(w_4)
+    assert t_4 == {}
+
+    w_5 = {"_nocorsika_sphere_radius": 16.0}
```

Comment:
Should this have the unit in order to make this test a bit more. meaningful?

Reply:
So your comment actually made me realize that the test is wrong. The input to `_initialize_sphere_parameters` is of type `{LST: 12.5 m, MST: 9.15 m, SST: 3 m}`. Fixed this.

@orel - good you have another look at the test?

Reply:
I must have had a reason for it, but all tests pass when adding an exception.

Could you have another look (sorry that it takes so many iterations).

Reply:
Oh that really confused me and cost 10 minutes. GitHub showed me changes only from some commits and not others I think and I saw mismatched test and code. 

It seems the test works now as expected (i.e., raises the exception), what else should be checked?

Reply:
OK, not sure about the github issues.

All what changed is that an Exception is raised now, as you suggested. This has been added, plus the corresponding test, see: https://github.com/gammasim/simtools/pull/733/commits/37c5821aba76b293adf59c532ee27adbeb41950c

Reply:
Yes, I agree tests pass and all is good now. We can merge in my opinion.